dashboard/app/helpers/openai_chat_helper.rb

```diff
@@ -0,0 +1,17 @@
+module OpenaiChatHelper
+  OPEN_AI_URL = "https://api.openai.com/v1/chat/completions"
+
+  def self.request_chat_completion(headers, data)
+    HTTParty.post(OPEN_AI_URL, headers: headers, body: data.to_json)
+  end
+
+  def self.get_chat_completion_response_message(response)
+    # Parse the response JSON and return the chat response message
+    if response.code == 200
+      response_body = JSON.parse(response.body)
+      return response_body['choices'][0]['message']
```

Comment:
To make sure I understand this (Ruby syntax always takes me for a trip), is `message` a string or a hash? And what would be returned here in the error case?

Reply:
`‘message’` is a hash and includes keys `role` and `content`. An example of the full response (parsed JSON) is:
```
{
  "id": "chatcmpl-123",
  "object": "chat.completion",
  "created": 1677652288,
  "choices": [{
    "index": 0,
    "message": {
      "role": "assistant",
      "content": "\n\nHello there, how may I assist you today?",
    },
    "finish_reason": "stop"
  }],
  "usage": {
    "prompt_tokens": 9,
    "completion_tokens": 12,
    "total_tokens": 21
  }
}
```
[Open AI API docuementation link](https://platform.openai.com/docs/api-reference/making-requests)

An example of an error case is If `messages` in the request body does not include both required keys `role` and `content`. If `messages` only includes `role`, then the response code returned would be 400 and the body would include an `error` which would include a descriptive `message` such as:
```
{"error"=>{"message"=>"'content' is a required property - 'messages.0'", "type"=>"invalid_request_error", "param"=>nil, "code"=>nil}}

Reply:
Thanks! Maybe one tangent...I would have expected the `status: :bad_request` part of what's returned in the error case below to be where you're actually sending the response (ie, `render`). Maybe just have this helper always return the content you want to respond with (which would be more consistent between your happy case and error case), and set the status where you're actually responding?

Reply:
Great point! I refactored so that if the error case is returned by OpenAI, the error is carried over by the controller. Thanks!